chart/templates/_helpers.yaml

```diff
@@ -198,51 +198,49 @@ If release name contains chart name it will be used as a full name.
   securityContext: {{- include "localContainerSecurityContext" .Values.dags.gitSync | nindent 4 }}
   env:
     {{- if .Values.dags.gitSync.sshKeySecret }}
-    - name: GIT_SSH_KEY_FILE
+    - name: GITSYNC_SSH_KEY_FILE
       value: "/etc/git-secret/ssh"
-    - name: GIT_SYNC_SSH
+    - name: GITSYNC_SSH
       value: "true"
     {{- if .Values.dags.gitSync.knownHosts }}
-    - name: GIT_KNOWN_HOSTS
+    - name: GITSYNC_SSH_KNOWN_HOSTS
       value: "true"
-    - name: GIT_SSH_KNOWN_HOSTS_FILE
+    - name: GITSYNC_SSH_KNOWN_HOSTS_FILE
       value: "/etc/git-secret/known_hosts"
     {{- else }}
-    - name: GIT_KNOWN_HOSTS
+    - name: GITSYNC_SSH_KNOWN_HOSTS
       value: "false"
     {{- end }}
     {{ else if .Values.dags.gitSync.credentialsSecret }}
-    - name: GIT_SYNC_USERNAME
+    - name: GITSYNC_USERNAME
       valueFrom:
         secretKeyRef:
           name: {{ .Values.dags.gitSync.credentialsSecret | quote }}
-          key: GIT_SYNC_USERNAME
-    - name: GIT_SYNC_PASSWORD
+          key: GITSYNC_USERNAME
```

Comment:
We can't just change what key we use like this, another breaking change.

Reply:
I think that this breaking change must be made by anyone who wants to use git-sync v4.0.0+, but we can keep git-sync tag =~v3.9 in chart's values.yaml.

but for those who put v4.0.0+ in their values file it's necessary to upgrade their credentialsSecret as well

Reply:
No, we don't have to change the key in the secret, just the env var we are mounting it to (line 265). That makes it a non-breaking change. Alternatively, we could support both.